pkg/controller/argocd/role.go

```diff
@@ -60,34 +47,57 @@ func newClusterRole(name string, rules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD)
 	}
 }
 
-func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) (*v1.Role, error) {
+// reconcileRoles will ensure that all ArgoCD Service Accounts are configured.
+func (r *ReconcileArgoCD) reconcileRoles(cr *argoprojv1a1.ArgoCD) (role *v1.Role, err error) {
+	if role, err := r.reconcileRole(applicationController, policyRuleForApplicationController(), cr); err != nil {
+		return role, err
+	}
+
+	if role, err := r.reconcileRole(dexServer, policyRuleForDexServer(), cr); err != nil {
+		return role, err
+	}
 
-	role := newRoleWithName(name, cr)
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: role.Name, Namespace: cr.Namespace}, role) //rbacClient.Roles(cr.Namespace).Get(context.TODO(), role.Name, metav1.GetOptions{})
-	roleExists := true
+	if role, err := r.reconcileRole(server, policyRuleForServer(), cr); err != nil {
+		return role, err
+	}
+
+	if role, err := r.reconcileRole(redisHa, policyRuleForRedisHa(), cr); err != nil {
+		return role, err
+	}
 
+	if _, err := r.reconcileClusterRole(applicationController,
+		policyRuleForApplicationControllerClusterRole(), cr); err != nil {
+		return nil, err
+	}
+
+	if _, err := r.reconcileClusterRole(server, policyRuleForServerClusterRole(), cr); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+// reconcileRole
+func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) (*v1.Role, error) {
+
+	role := newRole(name, policyRules, cr)
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: role.Name, Namespace: cr.Namespace}, role)
 	if err != nil {
 		if !errors.IsNotFound(err) {
-			return nil, fmt.Errorf("Failed to reconcile the role for the service account associated with %s : %s", name, err)
+			return nil, err
```

Comment:
I think this should be restored, note that the error message should start with a lower-case F ;-)